### PR TITLE
CB Add e2e coverage

### PR DESCRIPTION
**Pull request by kuhlmanp**
#### Description of changes
This adds several items to the CodeBuild setup including CodeCov updates and V8 supported coverage for e2e tests.

The CodeBuild updates include the function `_uploadCoverageLinux` in the shared-scripts.sh file. This setup utilizes the CodeCov Token secret, managed by SecretsManager and injected by the build.  It then gets the latest CodeCov uploader executable and verifies it via SHA. Once the verification is complete, the uploader API is called, automatically searching for coverage artifacts to upload. This update also includes an update to the codecov.yml file to conform to CodeCov validation.

Lastly, in this update we take advantage of the embedded ability of NodeJS to report coverage metrics: https://blog.npmjs.org/post/178487845610/rethinking-javascript-test-coverage.html. We then run the c8 tool to convert that coverage into something that both CodeCov and CodeBuild can consume. It also integrates these reports in CodeBuild directly.

Notes: currently, this mechanism is set to run tests from $TEST_SUITE and, in this PR, is set to run only the geo-add-b.test.ts file.

#### Description of how you validated changes
Manual build tests.

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/run_coverage_linux.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+version: 0.2
+env:
+  shell: bash
+  variables:
+    CLI_REGION: us-east-1
+    E2E_TEST_COVERAGE_DIR: node_v8_coverage
+  secrets-manager:
+    CODECOV_TOKEN: arn:aws:secretsmanager:us-east-1:052112422291:secret:testing/codecov/token-OB8isC
```
Comment (sdstolworthy) [inline codebuild_specs/run_coverage_linux.yml:8]: https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#secrets-manager-build-spec

Do we need the full ARN here? Is this secret stored in a different account? @awsluja any insight here? Putting a full ARN here feels wrong.

File: codebuild_specs/run_coverage_linux.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+phases:
+  build:
+    commands:
+      # you can provide a codebuild source version to use old cache and skip all other jobs :)
+      - export CODEBUILD_SOURCE_VERSION=00f31ea25fc27aa865198398e5014ea4d6a126d9
```
Comment (sdstolworthy) [inline codebuild_specs/run_coverage_linux.yml:13]: what does this reference? Does it need to be hardcoded?

File: codebuild_specs/run_e2e_tests_linux.yml [hunk 3]
```diff
@@ -26,19 +27,18 @@ phases:
   build:
     commands:
       # you can provide a codebuild source version to use old cache and skip all other jobs :)
+      # - export CODEBUILD_SOURCE_VERSION=00f31ea25fc27aa865198398e5014ea4d6a126d9
```
Comment (sdstolworthy) [inline codebuild_specs/run_e2e_tests_linux.yml:30]: delete dead comment

File: codecov.yml [hunk 4]
```diff
@@ -12,5 +12,5 @@ coverage:
     patch: off
     changes:
       default:
-        target: 80%
+        # target: 80%
```
Comment (sdstolworthy) [inline codecov.yml:15]: dead comment

**Comment by kuhlmanp** [comment]
pulled in too many changes from dev.

**Pull request closed without merging**